src/DefaultBuilder/src/BootstrapHostBuilder.cs

```diff
@@ -103,6 +118,16 @@ internal void RunConfigurationCallbacks()
             }
 
             _environment.ApplyConfigurationSettings(_configuration);
+
```

Comment:
Why do we `ApplyConfigurationSettings` twice when nothing is observing changes to `_environment` between these 2 calls.?

Reply:
This isn't new. ConfigureHostConfiguration callbacks to read environment variables (e.g. `DOTNET_` and `ASPNETCORE_`) that need to be reacted to before running ConfigureAppConfiguration callbacks to read config that's dependent on the environment name configured by the environment variables (e.g. `appsettings.{env.EnvironmentName}.json` and user secrets).

This is the last call to `ApplyConfigurationSettings` before the WebApplicationBuilder ctor exits, so it's to ensure app code in Program.cs can observe any changes made to the environment in appsettings mostly.

Reply:
After playing with this a bit I think we only need to read hosting configuration once at the beginning and blocking it after that. We should also get rid of our custom IWebHostEnvironment and should use the one configured in the `IHostBuilder.Properties[typeof(WebHostContext)]`

Reply:
Agreed!